hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -761,6 +764,11 @@ public AbfsRestOperation append(final String path, final byte[] buffer,
       requestHeaders.add(new AbfsHttpHeader(USER_AGENT, userAgentRetry));
     }
 
+    // Add MD5 Hash of request content as request header if feature is enabled
+    if (isChecksumValidationEnabled()) {
```

Comment:
In case of appends, as per REST API doc, server will fail: "If the two hashes do not match, the operation will fail with error code 400 (Bad Request)."
Are there indications in server error code response header to determine its due to MD5 mismatch and can get converted to AbfsInvalidChecksumException too ?

Reply:
ok, so Md5Mismatch is the stable error code to look for?

Reply:
Yes